cmake_uninstall.cmake.in

```diff
@@ -1,33 +1,40 @@
 if(NOT EXISTS "@CMAKE_BINARY_DIR@/install_manifest.txt")
-	message(FATAL_ERROR "Cannot find install manifest: @CMAKE_BINARY_DIR@/install_manifest.txt")
+        message(FATAL_ERROR "Cannot find install manifest: @CMAKE_BINARY_DIR@/install_manifest.txt")
 endif()
 
 file(READ "@CMAKE_BINARY_DIR@/install_manifest.txt" files)
 string(REGEX REPLACE "\n" ";" files "${files}")
 foreach(file ${files})
-	message(STATUS "Uninstalling $ENV{DESTDIR}${file}")
-	if(IS_SYMLINK "$ENV{DESTDIR}${file}" OR EXISTS "$ENV{DESTDIR}${file}")
-		exec_program(
-			"@CMAKE_COMMAND@" ARGS "-E rm -r \"$ENV{DESTDIR}${file}\""
-			OUTPUT_VARIABLE rm_out
-			RETURN_VALUE rm_retval
-			)
-		if(NOT "${rm_retval}" STREQUAL 0)
-			message(FATAL_ERROR "Problem when removing $ENV{DESTDIR}${file}")
-		endif()
-	else(IS_SYMLINK "$ENV{DESTDIR}${file}" OR EXISTS "$ENV{DESTDIR}${file}")
-		message(STATUS "File $ENV{DESTDIR}${file} does not exist.")
-	endif()
+        message(STATUS "Uninstalling $ENV{DESTDIR}${file}")
+        if(IS_SYMLINK "$ENV{DESTDIR}${file}" OR EXISTS "$ENV{DESTDIR}${file}")
+                if(${CMAKE_VERSION} VERSION_GREATER_EQUAL "3.17")
+                        exec_program(
+                                "@CMAKE_COMMAND@" ARGS "-E rm -r \"$ENV{DESTDIR}${file}\""
+                                OUTPUT_VARIABLE rm_out
+                                RETURN_VALUE rm_retval
+                                )
+                else(${CMAKE_VERSION} VERSION_LESS "3.17")
+                        exec_program(
+                                "@CMAKE_COMMAND@" ARGS "-E remove \"$ENV{DESTDIR}${file}\""
+                                OUTPUT_VARIABLE rm_out
+                                RETURN_VALUE rm_retval
+                                )
+                endif()
```

Comment:
Two things:

- You can use `${CMAKE_COMMAND}` instead of `@CMAKE_COMMAND@`

- `VERSION_GREATER_EQUAL` was added in CMake 3.7, so it is better to swap the two branches and have the first one using `VERSION_LESS`. Then you keep compatibility with older versions. And btw you don't need anything in the else().

Reply:
Also, We can squash this into the main "add uninstall target" patch...

Reply:
fixed this in a new commit